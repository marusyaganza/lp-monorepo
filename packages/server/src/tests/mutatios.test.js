const gql = require('graphql-tag')
const createTestServer = require('./helpers');
const {words, users} = require('./mocks/data');

const loginData = {email: 'test@test.com', password: 'password'};
const signUpData = {name: 'User', email: 'test@test.com', password: 'password', role: 'MEMBER'};

const loginMutation = gql`
mutation {
    login(input: {email: "member@member.com", password: "password"}) {
        id
        email
        name
        token
  }
}
`
const signUpMutation = gql`
mutation {
    signUp(input: {name: "User", email: "test@test.com", password: "password"}) {
        id
        email
        name
        token
  }
}
`

const createToken = jest.fn(() => 'token');

describe('mutations', () => {
  test('login with correct password', async () => {
    const findOne = jest.fn(() => users[0])
    const {mutate} = createTestServer({
    createToken,
      models: {
        User: {
          findOne,
        }
      }
    })

    const res = await mutate({mutation: loginMutation})
    expect(res).toMatchSnapshot();
    expect(findOne).toHaveBeenCalledWith({email: "member@member.com"});
    expect(createToken).toHaveBeenCalledWith(users[0]);
  });

  test('login with incorrect password', async () => {
    const findOne = jest.fn(() => null)
    const {mutate} = createTestServer({
    createToken,
      models: {
        User: {
          findOne,
        }
      }
    })

    const res = await mutate({mutation: loginMutation})
    expect(res).toMatchSnapshot();
  });
  
  test('signUp', async () => {
    const findOne = jest.fn(() => null);
    const createOne = jest.fn(() => ({...loginData, id: '2'}))
    const {mutate} = createTestServer({
    createToken,
      models: {
        User: {
          findOne,
          createOne,
        }
      }
    })

    const res = await mutate({mutation: signUpMutation})
    expect(res).toMatchSnapshot();
    expect(findOne).toHaveBeenCalledWith({email: "test@test.com"});
    expect(createOne).toHaveBeenCalledWith(signUpData);
  });
})