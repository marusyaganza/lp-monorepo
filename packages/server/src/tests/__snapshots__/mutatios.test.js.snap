// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations login with correct password 1`] = `
Object {
  "data": Object {
    "login": Object {
      "email": "member@member.com",
      "firstName": "User",
      "id": "1",
      "lastName": "Test",
      "token": "token",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutations login with incorrect password 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: email or password is incorrect],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutations signUp 1`] = `
Object {
  "data": Object {
    "signUp": Object {
      "email": "test@test.com",
      "firstName": "Hello World",
      "id": "2",
      "lastName": "Hello World",
      "token": "token",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
